src/m365/entra/commands/group/group-set.ts

```diff
@@ -0,0 +1,330 @@
+          }
+        }
+
+        if (args.options.ownerIds) {
+          const isValidGUIDArrayResult = validation.isValidGuidArray(args.options.ownerIds);
+          if (isValidGUIDArrayResult !== true) {
+            return `The following GUIDs are invalid for option 'ownerIds': ${isValidGUIDArrayResult}.`;
+          }
+        }
+
+        if (args.options.ownerUserNames) {
+          const isValidUPNArrayResult = validation.isValidUserPrincipalNameArray(args.options.ownerUserNames);
+          if (isValidUPNArrayResult !== true) {
+            return `The following user principal names are invalid for option 'ownerUserNames': ${isValidUPNArrayResult}.`;
+          }
+        }
+
+        if (args.options.memberIds) {
+          const isValidGUIDArrayResult = validation.isValidGuidArray(args.options.memberIds);
+          if (isValidGUIDArrayResult !== true) {
+            return `The following GUIDs are invalid for option 'memberIds': ${isValidGUIDArrayResult}.`;
+          }
+        }
+
+        if (args.options.memberUserNames) {
+          const isValidUPNArrayResult = validation.isValidUserPrincipalNameArray(args.options.memberUserNames);
+          if (isValidUPNArrayResult !== true) {
+            return `The following user principal names are invalid for option 'memberUserNames': ${isValidUPNArrayResult}.`;
+          }
+        }
+
+        if (args.options.visibility && !this.allowedVisibility.includes(args.options.visibility)) {
+          return `Option 'visibility' must be one of the following values: ${this.allowedVisibility.join(', ')}.`;
+        }
+
+        if (args.options.newDisplayName === undefined && args.options.description === undefined && args.options.visibility === undefined
+          && args.options.ownerIds === undefined && args.options.ownerUserNames === undefined && args.options.memberIds === undefined
+          && args.options.memberUserNames === undefined && args.options.mailNickname === undefined) {
+          return `Specify at least one of the following options: 'newDisplayName', 'description', 'visibility', 'ownerIds', 'ownerUserNames', 'memberIds', 'memberUserNames', 'mailNickname'.`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(
+      { options: ['id', 'displayName'] },
+      {
+        options: ['ownerIds', 'ownerUserNames', 'memberIds', 'memberUserNames'],
+        runsWhen: (args) => args.options.ownerIds || args.options.ownerUserNames || args.options.memberIds || args.options.memberUserNames
+      }
+    );
+  }
+
+  #initTypes(): void {
+    this.types.string.push('id', 'displayName', 'newDisplayName', 'description', 'mailNickname', 'ownerIds', 'ownerUserNames', 'memberIds', 'memberUserNames', 'visibility');
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    let groupId = args.options.id;
+    let ownerIds: string[] = [];
+    let memberIds: string[] = [];
```

Comment:
Is there a reason why we define these variables here?

Reply:
no